src/librustc_save_analysis/lib.rs

```diff
@@ -324,17 +328,40 @@ impl<'l, 'tcx: 'l> SaveContext<'l, 'tcx> {
                     let sub_span = self.span_utils.sub_span_for_type_name(path.span);
                     filter!(self.span_utils, sub_span, typ.span, None);
 
+                    let kind = match *trait_ref {
+                        Some(_) => ImplKind::Direct,
+                        None => ImplKind::Inherent,
+                    };
+
+                    let impl_id = self.make_impl_id(&kind);
+                    let span = self.span_from_span(sub_span.unwrap());
+
                     let type_data = self.lookup_ref_id(typ.id);
                     type_data.map(|type_data| {
                         Data::RelationData(Relation {
                             kind: RelationKind::Impl,
-                            span: self.span_from_span(sub_span.unwrap()),
+                            span: span.clone(),
                             from: id_from_def_id(type_data),
                             to: trait_ref
                                 .as_ref()
                                 .and_then(|t| self.lookup_ref_id(t.ref_id))
                                 .map(id_from_def_id)
                                 .unwrap_or(null_id()),
+                            id: impl_id,
+                        },
+                        Impl {
+                            id: impl_id,
+                            kind: kind,
+                            span: span,
+                            value: String::new(),     // ???
+                            parent: None,             // ??? enclosing module?
```

Comment:
Yes, this should be the enclosing module

Reply:
Ok i fix.

Reply:
@nrc sorry I feel stupid here. can I actually get the parent module? All other match arms of `ast::ImplKind` set `parent: None`. Is this case different?